app/views/directives/create-secret.html

```diff
@@ -1,363 +1,367 @@
+              mode: 'json',
+              theme: 'eclipse',
+              onChange: aceChanged,
+              rendererOptions: {
+                fadeFoldWidgets: true,
+                showPrintMargin: false
+              }
+            }" ng-model="newSecret.data.dockerConfig" class="create-secret-editor ace-bordered" id="dockerconfig-editor" required></div>
+            <div class="help-block">
+              File with credentials and other configuration for connecting to a secured image registry.
+            </div>
+            <div class="has-warning" ng-show="invalidConfigFormat">
+              <span class="help-block">
+                Configuration file should be in JSON format.
+              </span>
+            </div>
+          </div>
+        </div>
+        <div ng-if="('serviceaccounts' | canI : 'update') && !serviceAccountToLink">
           <div class="form-group">
-            <label class="required">Service Account</label>
-            <ui-select required ng-model="newSecret.pickedServiceAccountToLink">
-              <ui-select-match placeholder="Service Account Name">{{$select.selected}}</ui-select-match>
-              <ui-select-choices repeat="sa in (serviceAccountsNames | filter : $select.search)">
-                <div ng-bind-html="sa | highlight : $select.search"></div>
-              </ui-select-choices>
-            </ui-select>
+            <div class="checkbox">
+              <label>
+                <input type="checkbox" ng-model="newSecret.linkSecret">
+                Link secret to a service account.
+                <a href="{{'managing_secrets' | helpLink}}" target="_blank"><span class="learn-more-inline">Learn More&nbsp;<i class="fa fa-external-link" aria-hidden="true"></i></span></a>
+              </label>
+            </div>
+          </div>
+
+          <div ng-if="newSecret.linkSecret">
+            <div class="form-group">
+              <label class="required">Service Account</label>
+              <ui-select required ng-model="newSecret.pickedServiceAccountToLink">
+                <ui-select-match placeholder="Service Account Name">{{$select.selected}}</ui-select-match>
+                <ui-select-choices repeat="sa in (serviceAccountsNames | filter : $select.search)">
+                  <div ng-bind-html="sa | highlight : $select.search"></div>
+                </ui-select-choices>
+              </ui-select>
+            </div>
           </div>
         </div>
       </div>
     </div>
   </div>
-  <div class="buttons gutter-top-bottom">
-    <button class="btn btn-primary"
-            type="button"
-            ng-disabled="secretForm.$invalid || secretForm.$pristine || invalidConfigFormat"
-            ng-click="create()">Create</button>
-    <button class="btn btn-default"
-            type="button"
-            ng-click="cancel()">Cancel</button>
+  <div class="modal-footer">
+    <div class="buttons gutter-top-bottom">
+      <button class="btn btn-lg btn-default"
```

Comment:
This should not be `btn-lg` any longer.

@rhamilto Can you review the changes to the dialog? It looks like we might have lost some of the changes from #2735. You probably want the diff w/o whitespace: https://github.com/openshift/origin-web-console/pull/2689/files?w=1

Reply:
Because the [create secret directive gets used outside a modal](https://github.com/openshift/origin-web-console/blob/master/app/views/create-secret.html#L15-L20), we actually don't want to include any modal-specific markup in this directive.  I've [hacked the button layout for the modal with some CSS](https://github.com/openshift/origin-web-console/blob/784c0deb83d4ba9cc2479180f9abd26ad77281a5/app/styles/_core.less#L1004-L1015).

Reply:
@jhadvig Is there any reason you needed to change this file for this fix? I think it should be ok as-is in master